### DISCUSSION TITLE
Configure Zed for Astro

### DISCUSSION THREAD

**Discussion by nickytonline** [zed-industries/zed#6640] [Feature Requests]
Hi all!

I know you can configure different LSPs in the Zed user settings, e.g.

```
  "lsp": {
    "rust-analyzer": {
      "initialization_options": {
        "checkOnSave": {
          "command": "clippy"
        }
      }
    }
  }
```

I'm curious if anyone knows how to add a new LSP server that does not ship with Zed to the user settings, specifically Astro. I know Zed uses tree-sitter, so I'm wondering how I can load a config to support *.astro files.

I see there is a proposal to get Astro support as well, https://github.com/zed-industries/community/issues/600

There is https://github.com/virchau13/tree-sitter-astro which works in Neovim, but I can't figure out how to load this tree-sitter via the config in Zed.

Thanks in advance!

**Comment by CarCanic** [comment]
Would love help too...

**Comment by ricardo-rp** [comment]
If there's a way to load new LSPs, it would be amazing to see that in the docs. 
If there's currently not, it would also be amazing to see that in the docs 😬

**Comment by SomeoneToIgnore** [comment]
There's an ongoing work on adding Astro support: https://github.com/zed-industries/zed/pull/6896
and also an ongoing work on generic language support, tracked here: https://github.com/zed-industries/zed/issues/7096

The latter is closing to its first public prototype, so not sure we want to add anything to docs before it comes out and sure have to update them afterwards.
